cmd/argo/commands/convert.go

```diff
@@ -0,0 +1,170 @@
+package commands
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/yaml"
+
+	"github.com/argoproj/argo-workflows/v3/cmd/argo/commands/common"
+	wf "github.com/argoproj/argo-workflows/v3/pkg/apis/workflow"
+	fileutil "github.com/argoproj/argo-workflows/v3/util/file"
+	jsonpkg "github.com/argoproj/argo-workflows/v3/util/json"
+	"github.com/argoproj/argo-workflows/v3/workflow/convert"
+)
+
+func NewConvertCommand() *cobra.Command {
+	var (
+		output = common.EnumFlagValue{
+			AllowedValues: []string{"yaml", "json"},
+			Value:         "yaml",
+		}
+	)
+
+	command := &cobra.Command{
+		Use:   "convert FILE...",
+		Short: "convert workflow manifests from legacy format to current format",
+		Long:  "Converts singular schedule, mutex and semaphore to the new plural version",
+		Example: `
+# Convert manifests in a specified directory:
+
+  argo convert ./manifests
+
+# Convert a single file:
+
+  argo convert workflow.yaml
+
+# Convert from stdin:
+
+  cat workflow.yaml | argo convert -`,
+		Args: cobra.MinimumNArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runConvert(cmd.Context(), args, output.String())
+		},
+	}
+
+	command.Flags().VarP(&output, "output", "o", "Output format. "+output.Usage())
+	command.Flags().BoolVar(&common.NoColor, "no-color", false, "Disable colorized output")
+
+	return command
+}
+
+var yamlSeparator = regexp.MustCompile(`\n---`)
```

Comment:
Could this be `const`?
```suggestion
const yamlSeparator = regexp.MustCompile(`\n---`)
```

It's only used once below so I would even suggest moving it into the function. Plus there's the "Split YAML documents" comment to describe it